test/csuite/wt3120_filesys/main.c

```diff
@@ -52,11 +53,17 @@ main(int argc, char *argv[])
  * in the test/csuite directory.
  */
 #define WT_FAIL_FS_LIB "ext/test/fail_fs/.libs/libwiredtiger_fail_fs.so"
-    if ((p = getenv("top_builddir")) == NULL)
-        p = "../../build_posix";
+    /* Get the git top level directory and append build_posix. */
+    fp = popen("git rev-parse --show-toplevel", "r");
+    ret = fscanf(fp, "%s", buf);
```

Comment:
`fscanf` is subject to buffer overrun problems.  It's relatively safe in this context, but I would get in the habit of writing secure code.  You could use format `"%1023s"`, but it's not that maintainable (what if someone reduces the size of buf without changing the number?).  I think the best bet is to avoid `fscanf` in this case, and try:
`fgets(buf,sizeof(buf),fp);`. It returns `NULL` on error.